### DISCUSSION TITLE
tuya-uncover: find local keys, uuid and used dp's for devices

### DISCUSSION THREAD

**Discussion by blakadder** [make-all/tuya-local#908] [General]
[tuya-uncover](https://github.com/blakadder/tuya-uncover) is a python script to find device local keys, uuid and used dp's. 

It requires only the login and password for the app and works on a number of Tuya OEM apps.

**Comment by make-all** [comment]
Interesting, this is the first method I've heard of that works with OEM apps without a rooted device.  Maybe it will be useful for some of the devices that do not have full functionality under the generic Tuya branded apps (there was a vacuum cleaner a while back that the user gave up on before all the data was collected to make a config, because they didn't want to lose the map functionality).

**Reply by blakadder** [reply]
I've went down that rabbit hole primarily for a Proscenic vacuum that was not really useful in the Tuya app because of a nonstandard map format. This way I can still use the app for features that are missing from HA (mapping and zoning) and integrate it in HA. 

If I could figure out how to display the vacuum maps as an image it would be perfect!
